client/blocks/site/index.jsx

```diff
@@ -116,6 +117,11 @@ class Site extends React.Component {
 			'is-compact': this.props.compact,
 		} );
 
+		const isPublicComingSoon =
+			isEnabled( 'coming-soon-v2' ) &&
+			! this.props.site.is_private &&
```

Comment:
You can possibly drop the is private check here - these settings should be decoupled.